### PR TITLE
LPD-22785 Make “Date and Time” and "Date" types available to be source for filters

### PR DESCRIPTION
**Pull request by markocikos**
### References

- Parent Story: [LPD-19647 Fix all the things](https://issues.liferay.com/browse/LPD-19647)

### What is the goal of this PR?

The goal is to support fields with `date-time` format. For this, we need to tweak code in several places, see inline comments in code.

This PR does **not** include Playwright tests, I'll add them in a separate task. This is somewhat blocked by https://github.com/brianchandotcom/liferay-portal/pull/148917

### What does it look like?

In screenshots below, `dateTest` is in `date` format, while `dateCreated` is in `date-time` format:

image

image

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/filters/Filters.tsx [hunk 1]
```diff
@@ -597,7 +597,8 @@ function Filters({fdsFilterClientExtensions, fdsView, namespace}: IProps) {
 					item.type === EFieldType.STRING &&
 					!item.format) ||
 				(filterType === EFilterType.DATE_RANGE &&
-					item.format === EFieldFormat.DATE)
+					(item.format === EFieldFormat.DATE ||
+						item.format === EFieldFormat.DATE_TIME))
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/filters/Filters.tsx:601]: This includes `data-time` format fields in DSM UI

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java [hunk 2]
```diff
@@ -644,7 +646,10 @@ private JSONArray _getFiltersJSONArray(
 					return JSONUtil.put(
 						"active", hasPreloadedData
 					).put(
-						"entityFieldType", FDSEntityFieldTypes.DATE
+						"entityFieldType",
+						Objects.equals(type, "date") ?
+							FDSEntityFieldTypes.DATE :
+								FDSEntityFieldTypes.DATE_TIME
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java:652]: Forwarding `date-time` fields as new entity type. My initial plan was to simply save `date-time` fields as `date` type  to DSM persistence, but it turns out `date-time` fields require different odata string format, so we need to tell them apart from `date` fields.

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/controls/filters/implementation/DateRangeFilter.tsx [hunk 3]
```diff
@@ -65,7 +66,7 @@ const getIsoString = ({
 
 	const dateISOString = date.toISOString();
 
-	if (entityFieldType === 'date') {
+	if (entityFieldType === EEntityFieldType.DATE) {
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/controls/filters/implementation/DateRangeFilter.tsx:69]: On this line, there is a fork in odata formats. We make the `.split('T')[0]` for the `date` entity type, and leave ISO-format string as it is for the `date-time` format.

**Comment by markocikos** [comment]
ci:test:relevant

**Review by juanjofgliferay — APPROVED** [review]
LGTM :rocket:

**Comment by juanjofgliferay** [comment]
@markocikos Just curious about the change of `components` folder by `controls` :thinking:

**Comment by markocikos** [comment]
> Just curious about the change of components folder by controls

@juanjofgliferay `components` has a special meaning, in the guidelines we should really add one of these days :sweat_smile: . It should contains stateless, reusable components. `controls` is simply a subdirectory for organization purposes, that can be anything.

**Comment by markocikos** [comment]
ci:forward

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java (hunk 2 above)
Comment (dsanz) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java:652]: yes!

**Pull request closed without merging**